### PR TITLE
NuGet prerelease package updater

### PR DESCRIPTION
**Pull request by zengin**
Dependabot doesn't look like to be implementing the feature for updating preview packages in the near future.

According to [this comment](https://github.com/dependabot/dependabot-core/issues/1842#issuecomment-624529062), they only allow updating from a production release to a preview release if their main versions are the same, e.g.
```
from 1.1.0 to 1.1.0.preview
```
which doesn't seem to fit the SemVer spec. See 11.3 [here](https://semver.org/#spec-item-11) which basically says
```
1.1.0.preview < 1.1.0
```

Anyways, my quick fix here is a GitHub action and an accompanying Powershell script, which fetches the latest NuGet packages for `Microsoft.Graph.Beta` and `Microsoft.Graph.Auth` libraries regardless of their prerelease status. No intelligence is built into understanding the version numbers, I simply grab the latest package.

I also use `peter-evans/create-pull-request@v3` GitHub action for creating a PR. It handles all the git operations I want:
- No-op if there is no changes
- Auto branching, committing and pushing
- Specific unique branch naming (timestamp suffix)

I reformatted the csproj file to set the baseline and decreased the version numbers to give this a go in the main repo as well. 

Action can be seen in action here: https://github.com/zengin/msgraph-sdk-raptor/pull/4

### CODE REVIEW

File: .github/workflows/prerelease-package-updater.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          signoff: false
+          branch: prerelease-updates
+          branch-suffix: timestamp
+          delete-branch: true
+          title: 'Updated prerelease dependencies'
+          body: |
+            Please review if only prerelease packages are updated to newer versions
+          labels: |
+            prerelease update
+          reviewers: zengin
```
Comment (zengin) [inline .github/workflows/prerelease-package-updater.yaml:40]:
```suggestion
          reviewers: |
            zengin
            baywet
```

**Review by nikithauc — APPROVED** [review]

**Review by baywet — APPROVED** [review]
If only they had a PowerShell SDK ;-)
Out of curiosity, is there any reason you didn't use the [dotnet cli](https://stackoverflow.com/a/59041841/3808675) to search for outdated packages?

**Comment by zengin** [comment]
@baywet, parsing that output didn't look clean compared to NuGet JSON response. I wish there was a way to directly update a specific package, something like `dotnet update <csproj> package Microsoft.Graph.Beta --prerelease`.

I gave this a second thought after your comment and this seems to be a cleaner solution:
```
dotnet remove .\msgraph-sdk-raptor-compiler-lib\msgraph-sdk-raptor-compiler-lib.csproj package Microsoft.Graph.Beta
dotnet add .\msgraph-sdk-raptor-compiler-lib\msgraph-sdk-raptor-compiler-lib.csproj package Microsoft.Graph.Beta --prerelease
```

Please let me know if you know an even cleaner solution with the cli.

**Comment by zengin** [comment]
marking this as draft while I test couple of things..

**Review by baywet — APPROVED** [review]

**Pull request merged**